numpy/core/arrayprint.py

```diff
@@ -641,7 +635,8 @@ def fillFormat(self, data):
 
         if self.exp_format:
             self.large_exponent = 0 < min_val < 1e-99 or max_val >= 1e100
-            self.max_str_len = 8 + self.precision
+            pad_sign = self.sign or _nc.any(non_zero < 0)
+            self.max_str_len = 6 + self.precision + pad_sign
```

Comment:
The simple answer is that there appeared to be an unnecessary space.

The more complex answer is that actually I'm not sure why that extra space was there... I'll take another look.